src/PackageCompiler.jl

```diff
@@ -686,7 +686,17 @@ function create_sysimg_from_object_file(object_files::Vector{String},
     end
     mkpath(dirname(sysimage_path))
     # Prevent compiler from stripping all symbols from the shared lib.
-    o_file_flags = Sys.isapple() ? `-Wl,-all_load $object_files` : `-Wl,--whole-archive $object_files -Wl,--no-whole-archive`
+    if Sys.isapple()
+        cltools_version_cmd = `pkgutil --pkg-info=com.apple.pkg.CLTools_Executables`
+        cltools_version = match(r"version: (.*)\n", readchomp(cltools_version_cmd))[1]
```

Comment:
This should handle the case that `pkgutil` is unable to locate `com.apple.pkg.CLTools_Executables`

Reply:
Good call. Do you have any thoughts on what version we should assume under this failure case? I don't have a good feel for whether it's more likely that a system has been upgraded to the latest version or not, but given how Apple ties Xcode version to macOS version (and professional dev envs are eventually going to force OS updates) it may be safest to default to 15+ (i.e. to apply the flags).

Reply:
Unfortunately it looks like there's a syntax ambiguity so that a lot of linkers interpret this flag as `-ld_classic` i.e. link against `libd_classic.so`, which will obviously fail.

I'm not sure how far back we have to go before that becomes an issue. It also sounds like we might have to worry about using a different compiler _not_ from XCode CLT.

Could we execute the compiler binary directly and ask it for its version directly?

Reply:
My biggest concern is that this patch assumes that the user is using the XCode CLT to compile on macOS, which is not necessarily true

By "execute the compiler binary directly", I mean a combination of either:
 - Modifying [get_compiler_cmd](https://github.com/JuliaLang/PackageCompiler.jl/blob/8e579083b1142dab26f268d77650e680cad6b4b5/src/PackageCompiler.jl#L151) to inspect the path to the binary and see that it's provided by Xcode, and/or
  - Execute `run_compiler(`` --version``)` to determine whether we're running the Xcode-provided compiler and which version.

Reply:
I've tinkered with this a bit and am still having a bit of trouble. 

I've tried getting the path to the compiler with
```
base_dir = dirname(sysimg_source_path)
compiler_source_path = joinpath(base_dir, "compiler", "compiler.jl")
```
Printing `compiler_source_path` results in `/Users/jaycook/.julia/juliaup/julia-1.8.5+0.aarch64.apple.darwin14/share/julia/base/compiler/compiler.jl`. I was hoping more for something like `/Library/Developer/CommandLineTools/usr/bin`, where I could check for a string like "xcode" or "CommandLineTools".

`run_compiler()`'s [underlying call to `run()`](https://github.com/JuliaLang/PackageCompiler.jl/blob/8e579083b1142dab26f268d77650e680cad6b4b5/src/PackageCompiler.jl#L200) does not appear to return the result of the command that's run. For example, `print(run_compiler(`` --version``))` results in `Process(``gcc --version``, ProcessExited(0))`.

If you're so inclined, do you care to toss up a commit for the implementation that you have in mind?